Source/CBL_ForestDBStorage.mm

```diff
@@ -992,8 +998,11 @@ - (CBL_Revision*) addDocID: (NSString*)inDocID
         return (CBLStatus)status;
     }];
 
-    if (CBLStatusIsError(*outStatus))
+    if (CBLStatusIsError(*outStatus)) {
+        if (outError && !*outError)
```

Comment:
!*outError check is required here to not overriding the validation error from line 972.